File: algorithms/levenshtein-distance/go/levenshtein-distance.go
package main

import "fmt"

func min(args ...int) int {
	minimum := args[0]

	for _, i := range args {
		if minimum > i {
			minimum = i
		}
	}

	return minimum
}

func levenshteinDistancePrivate(one, two string, len1, len2 int) int {
	if len1 == 0 {
		return len2
	}

	if len2 == 0 {
		return len1
	}

	cost := 0

	if one[len1-1] != two[len2-1] {
		cost = 1
	}

	return min(levenshteinDistancePrivate(one, two, len1-1, len2)+1, levenshteinDistancePrivate(one, two, len1, len2-1)+1, levenshteinDistancePrivate(one, two, len1-1, len2-1)+cost)
}

func levenshteinDistance(one, two string) int {
	return levenshteinDistancePrivate(one, two, len(one), len(two))
}

func main() {
	one := "Hello"
	two := "Jello"

	distance := levenshteinDistance(one, two)

	fmt.Println(distance)
}

File: algorithms/levenshtein-distance/python/levenshtein-distance.py
def levenshtein_distance_private(one, two, len1, len2):

	if len1 == 0:
		return len2

	if len2 == 0:
		return len1

	cost = 0

	# testa o último caractere
	if one[len1 - 1] != two[len2 - 1]:
		cost = 1

	return min(levenshtein_distance_private(one, two, len1 - 1, len2) + 1,
				levenshtein_distance_private(one, two, len1, len2 - 1) + 1,
				levenshtein_distance_private(one, two, len1 - 1, len2 - 1) + cost)

def levenshtein_distance(one, two):
    return levenshtein_distance_private(one, two, len(one), len(two))
    
one, two = 'Hello', 'Jello'
print(levenshtein_distance(one, two))